sdk/include/ndk/rtltypes.h

```diff
@@ -410,7 +410,11 @@ extern BOOLEAN NTSYSAPI NLS_MB_OEM_CODE_PAGE_TAG;
 #define RTL_CONSTANT_LARGE_INTEGER(quad_part) { .QuadPart = (quad_part) }
 #define RTL_MAKE_LARGE_INTEGER(low_part, high_part) { .LowPart = (low_part), .HighPart = (high_part) }
 #else
+#ifdef NONAMELESSUNION
+C_ASSERT(FIELD_OFFSET(LARGE_INTEGER, u.LowPart) == 0);
```

Comment:
I'm wondering then how the ".LowPart" stuff from line 411 could still work if NONAMELESSUNION is defined.

Reply:
It wouldn't. We just haven't attempted to compile in a configuration that supports modern C++, but disallows nameless unions.

Reply:
Note that there's no reason that NONAMELESSUNION should *ever* be defined, for any of our use cases. All compatible versions of msvc and gcc that we use for compiling reactos DO support nameless unions. I'm baffled as to what the F* AppVeyor is doing to cause that. Hence this test commit: if it compiles successfully, it means my assumption is correct, then I'll investigate which settings they are using that could possibly cause that.

Reply:
Simple answer:
https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=define+NONAMELESSUNION
These modules, if using the macros, would fail compiling.

Reply:
Oh wow that's ugly... I didn't realize there's actual components that force-enable that without it being a side-effect of using an old compiler. Okay, yeah, then we have to reconsider this: it may NOT be worth keeping the more readable version, if it's hidden behind so many conditionals...